pallets/pallet-bonded-coins/src/tests/transactions/create_pool.rs

```diff
@@ -1 +1,251 @@
+use frame_support::{assert_err, assert_ok};
+use frame_system::{pallet_prelude::OriginFor, RawOrigin};
+use pallet_assets::{Error as AssetsPalletErrors, Event as AssetsPalletEvents};
+use sp_runtime::{ArithmeticError, BoundedVec};
+use sp_std::ops::Sub;
 
+use crate::{
+	mock::runtime::*, mock::*, types::PoolStatus, Event as BondingPalletEvents, NextAssetId, Pools, TokenMetaOf,
+};
+
+// create_pool tests
+
+#[test]
+fn single_currency() {
+	let initial_balance = 100_000_000_000_000_000u128;
+	ExtBuilder::default()
+		.with_native_balances(vec![(ACCOUNT_00, initial_balance)])
+		.with_collaterals(vec![DEFAULT_COLLATERAL_CURRENCY_ID])
+		.build()
+		.execute_with(|| {
+			assert_eq!(NextAssetId::<Test>::get(), 0);
+			assert_eq!(initial_balance, Balances::free_balance(ACCOUNT_00));
+
+			let origin = RawOrigin::Signed(ACCOUNT_00).into();
+			let curve = get_linear_bonding_curve_input();
+
+			let bonded_token = TokenMetaOf::<Test> {
+				name: BoundedVec::truncate_from(b"Bitcoin".to_vec()),
+				symbol: BoundedVec::truncate_from(b"btc".to_vec()),
+				min_balance: 1,
+			};
+
+			assert_ok!(BondingPallet::create_pool(
+				origin,
+				curve,
+				DEFAULT_COLLATERAL_CURRENCY_ID,
+				BoundedVec::truncate_from(vec![bonded_token]),
+				10,
+				true
+			));
+
+			let pool_id = calculate_pool_id(&[0]);
+
+			let details = Pools::<Test>::get(&pool_id).unwrap();
+
+			assert!(details.is_owner(&ACCOUNT_00));
+			assert!(details.is_manager(&ACCOUNT_00));
+			assert!(details.transferable);
+			assert_eq!(details.state, PoolStatus::Locked(Default::default()));
```

Comment:
I think we want to math against the exact state, and not use `Default::default()`. If tomorrow the default canges, we would fail to spot the change in the tests.

Reply:
yeah fair, 0423e17f1e8d35574f3912fa72c38e21e4c9c745